sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -298,3 +299,36 @@ case class StringLength(child: Expression) extends UnaryExpression with ExpectsI
 
   override def prettyName: String = "length"
 }
+
+/**
+ * Returns string, with the first letter of each word in uppercase,
+ * all other letters in lowercase. Words are delimited by whitespace.
+ */
+case class InitCap(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+  override def dataType: DataType = StringType
+
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+
+  override def eval(input: InternalRow): Any = {
+    val string = child.eval(input)
+    if (string == null) {
+      null
+    }
+    else if (string.toString.length == 0) {
+      return UTF8String.fromString(string.toString)
+    }
+    else {
+      val str: String = string.toString
+      val sb = new StringBuilder
+      sb.append(str)
+      sb.setCharAt(0, sb(0).toUpper)
+
+      for (i <- 1 until sb.length) {
+        if (sb(i - 1).equals(' ')) {
+          sb.setCharAt(i, sb(i).toUpper)
+        }
```

Comment:
Please have a look on https://gist.github.com/tarekauel/0196ed51aa43a23afc12

I know that `str` is very big, but even with a small String like `str = "aaaaaaa"` it's still faster (example result: `A: 5  B: 1`)

Why does it waste memory?